### PR TITLE
feat(rtc): reconnect when connection got closed

### PR DESCRIPTION
**Pull request by reinos**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our conventions](../CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?
When the connection of the RTC got closed (e.g. connection lost or server down), the connection will not be restored if possible.

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The RTC will reconnect in case of unexpected closed event.

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/api/realtime/realTime.interfaces.ts [hunk 1]
```diff
@@ -39,7 +40,7 @@ export interface IWebSocket {
   /**
    * `close` from WebSocket
    */
-  close(): void;
+  close(code?: number): void;
```
Comment (veger) [inline src/api/realtime/realTime.interfaces.ts:43]: `code?` is optional param in TS?

Comment (reinos) [reply src/api/realtime/realTime.interfaces.ts:43]: Yes 👍

File: src/api/realtime/realTimeModule.ts [hunk 2]
```diff
@@ -250,15 +253,35 @@ export class RealTimeModule implements IModule {
...
+      .then(() => websocket.start(this.websocket, () => this.tokenManager.token().toPromise(), reconnection));
   }
 
+  /**
+   * Reconnect the websocket when it got unexpectedly closed
+   *
+   * @internal
+   */
+  private reconnect(event: CloseEvent) {
+    if (event instanceof CloseEvent && event.code !== 4999) {
```
Comment (veger) [inline src/api/realtime/realTimeModule.ts:273]: magic number: `4999`

Comment (veger) [reply src/api/realtime/realTimeModule.ts:273]: ~and according to https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent the full range of 4000-4999 is for custom events...~ nevermind I later saw _you_ send the `4999`... so I guess using some constant for this magic number indeed can help :smile:

**Review by veger — APPROVED** [review]

File: src/api/realtime/realTimeModule.ts (hunk 2 above)
Comment (reinos) [reply src/api/realtime/realTimeModule.ts:273]: 😊  yes will make a constant for that, should be better.

**Pull request merged**